modules/api/src/main/java/org/apache/ignite/table/criteria/Criteria.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.table.criteria;
+
+import java.util.Arrays;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents a predicate for filtering table entries.
+ *
+ * <pre><code>
+ *      public ClosableCursor&lt;Product&gt; uncategorizedProducts() {
+ *         return products.recordView(Product.class).queryCriteria(null, columnValue(&quot;category&quot;, nullValue()));
+ *      }
+ * </code></pre>
+ *
+ * @see CriteriaQuerySource
+ */
+public interface Criteria {
+    /**
+     * Accept the visitor with the given context.
+     *
+     * @param <C> context type.
+     * @param v visitor.
+     * @param context context of visit.
+     */
+    <C> void accept(CriteriaVisitor<C> v, @Nullable C context);
+
+    /**
+     * Creates a predicate that tests whether the column value is equal to the given condition.
+     *
+     * <p>For example:
+     * <pre>
+     *     columnValue(&quot;category&quot;, equalTo(&quot;toys&quot;))
+     * </pre>
+     *
+     * @param columnName Column name.
+     * @param condition Target condition.
+     * @return The created expression instance.
+     */
+    static Expression columnValue(String columnName, Condition condition) {
+        var oldElements = condition.getElements();
```

Comment:
using `var` for such caseses are forbidden

Reply:
Fixed. Checked all other places